src/core_codemods/file_resource_leak.py

```diff
@@ -157,59 +172,118 @@ class ResourceLeakFixer(
     def __init__(
         self,
         context: CodemodContext,
-        leaked_assigned_resources: list[
-            tuple[
-                cst.IndentedBlock | cst.Module,
-                cst.SimpleStatementLine,
-                cst.Assign | cst.AnnAssign,
-                cst.Call,
-            ]
+        leaked_assigned_resources: dict[
+            cst.IndentedBlock | cst.Module,
+            list[
+                tuple[
+                    cst.SimpleStatementLine,
+                    cst.Assign | cst.AnnAssign,
+                    cst.Call,
+                ]
+            ],
         ],
     ):
         super().__init__(context)
         self.leaked_assigned_resources = leaked_assigned_resources
         self.changes: list[Change] = []
 
-    def leave_Module(self, original_node: cst.Module, updated_node) -> cst.Module:
-        result = original_node
-        # TODO for now it assumes no dependent resources, it won't do anything if one exists
-        for (
-            block,
-            stmt,
-            assignment,
-            resource,
-        ) in self.leaked_assigned_resources:
-            index = block.body.index(stmt)
+    def _is_fixable(
+        self, block, index, named_targets, other_targets
+    ) -> bool:  # pylint: disable=too-many-arguments
+        # assigned to something that is not a Name?
+        if other_targets:
+            return False
+        # yield, returned, argument of a call, referenced outside of block
+        name_escapes_partial = partial(
+            self._name_escapes_scope, block=block, index=index
+        )
+        # is closed?
+        name_condition = map(
+            # pylint: disable-next=cell-var-from-loop
```

Comment:
hmmm either this pylint warning is worth listening to or it's a pylint bug? Either way, maybe a list-comp is easier to read here?

Reply:
I wanted to avoid building the list since the result would just be fed into `all`. I don't exactly remember the issue with the warning was. It may be invalid by now. I'll try to remove and see if ti still complains.